system/include/emscripten/bind.h

```diff
@@ -1120,8 +1120,8 @@ namespace emscripten {
             BaseSpecifier::template verify<ClassType>();
 
             auto _getActualType = &getActualType<ClassType>;
-            auto upcast = BaseSpecifier::template getUpcaster<ClassType>();
-            auto downcast = BaseSpecifier::template getDowncaster<ClassType>();
+            void (*upcast)(void) = (void (*)(void))BaseSpecifier::template getUpcaster<ClassType>();
```

Comment:
sgtm

Declared it local to that function because we only use it there, made things look nicer